test/boost/hint_test.cc

```diff
@@ -33,6 +33,9 @@ std::ostream& operator<<(std::ostream& out, const replay_position& p) {
 
 template <>
 struct fmt::formatter<db::hints::sync_point::host_id_or_addr> {
+    constexpr static auto parse(format_parse_context& ctx) {
+        return ctx.begin();
+    }
```

Comment:
I tried reading the documentation of `fmt` and couldn't find a good description of what the function is supposed to return, only to either look at existing implementations in `fmt` or derive from an existing implementation of a formatter (`fmt::formatter<string_view>` seems to be a common choice in our codebase).

However, given that the pattern with `return ctx.begin()` is also common in our codebase and @tchaikov approved, then it LGTM as well.

Reply:
it's like [`std::from_chars()`](https://en.cppreference.com/w/cpp/utility/from_chars) -- it parses the format string until it reaches `}`, which marks the end of the placeholder of the current formatted argument. but if this function reaches the end of the range before hitting `}`, that's an issue in the format specifier. if everything goes fine, this function returns an iterator pointing to `}`. so that {fmt} can move foward with the rest of the format string.